core/dbt/deps/resolver.py

```diff
@@ -74,6 +75,10 @@ def update_from(self, src: List[PackageSpec]) -> None:
                 pkg = TarballUnpinnedPackage.from_contract(contract)
             elif isinstance(contract, GitPackage):
                 pkg = GitUnpinnedPackage.from_contract(contract)
+            elif isinstance(contract, PrivatePackage):
+                raise DbtInternalError(
```

Comment:
I don't think this should be a DbtInternalError if it can be caused by user-configuration and is resolvable given user action